MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -4165,6 +4251,30 @@ static uint32_t mvkGetEntryProperty(io_registry_entry_t entry, CFStringRef prope
 	mvkRemoveFirstOccurance(_awaitingTimelineSem4s, make_pair(sem4, value));
 }
 
+MVKAccelerationStructure* MVKDevice::addAccelerationStructure(MVKAccelerationStructure* accStruct) {
+    std::pair<uint64_t, MVKAccelerationStructure*> accStructMemoryPair = std::make_pair(_nextValidAccStructureAddress, accStruct);
+    _gpuAccStructAddressMap.insert(accStructMemoryPair);
+    accStruct->setDeviceAddress(_nextValidAccStructureAddress);
+    _nextValidAccStructureAddress += accStruct->getMTLSize();
+    return accStruct;
+}
+
+void MVKDevice::removeAccelerationStructure(MVKAccelerationStructure* accStruct) {
+    std::unordered_map<uint64_t, MVKAccelerationStructure*>::iterator accStructIt = _gpuAccStructAddressMap.find(accStruct->getDeviceAddress());
+    uint64_t addressOffset = accStructIt->second->getMTLSize();
+    _gpuAccStructAddressMap.erase(accStructIt);
+    
+    // This can lead to fragmentation over time, so I'll just push all keys after this back
+    // This, however is also another performance issue
```

Comment:
I wonder, then, if it wouldn't be better to use an ordered (RB-tree-based) `std::map<K, V>` instead.